src/python/grpcio_tests/tests_aio/unit/interceptor_test.py

```diff
@@ -685,6 +685,110 @@ class Interceptor(aio.UnaryUnaryClientInterceptor):
+            interceptors=self._interceptors)
+
+    async def tearDown(self) -> None:
+        self._server.stop(None)
+
+    async def test_invalid_interceptor(self):
+        class InvalidInterceptor:
+            """Just an invalid Interceptor"""
+
+        with self.assertRaises(aio.AioRpcError):
+            server_target, _ = await start_test_server(
+                interceptors=(InvalidInterceptor(),))
+            channel = aio.insecure_channel(server_target)
+            multicallable = channel.unary_unary(
+                '/grpc.testing.TestService/UnaryCall',
+                request_serializer=messages_pb2.SimpleRequest.SerializeToString,
+                response_deserializer=messages_pb2.SimpleResponse.FromString)
+            call = multicallable(messages_pb2.SimpleRequest())
+            await call
+
+    async def test_executed_right_order(self):
+        self._record.clear()
+        async with aio.insecure_channel(self._server_target) as channel:
+            multicallable = channel.unary_unary(
+                '/grpc.testing.TestService/UnaryCall',
+                request_serializer=messages_pb2.SimpleRequest.SerializeToString,
+                response_deserializer=messages_pb2.SimpleResponse.FromString)
+            call = multicallable(messages_pb2.SimpleRequest())
+            response = await call
+
+            # Check that all interceptors were executed, and were executed
+            # in the right order.
+            self.assertSequenceEqual(['log1:intercept_service',
+                                      'log2:intercept_service',], self._record)
+            self.assertIsInstance(response, messages_pb2.SimpleResponse)
+
+    async def test_apply_different_interceptors_by_metadata(self):
+        async with aio.insecure_channel(self._server_target) as channel:
+            multicallable = channel.unary_unary(
+                '/grpc.testing.TestService/UnaryCall',
+                request_serializer=messages_pb2.SimpleRequest.SerializeToString,
+                response_deserializer=messages_pb2.SimpleResponse.FromString)
+            self._record.clear()
+            metadata = (('key', 'value'),)
+            call = multicallable(messages_pb2.SimpleRequest(),
+                                 metadata=metadata)
+            await call
+            self.assertSequenceEqual(['log1:intercept_service',
+                                      'log2:intercept_service',],
+                                     self._record)
+
+            self._record.clear()
+            metadata = (('key', 'value'), ('secret', '42'))
+            call = multicallable(messages_pb2.SimpleRequest(),
+                                 metadata=metadata)
+            await call
+            self.assertSequenceEqual(['log1:intercept_service',
+                                      'log3:intercept_service',
+                                      'log2:intercept_service',],
+                                     self._record)
+
```

Comment:
Should we add one more test case that the interceptor returns a different method handler?

Reply:
I couldn't understand this one, can you give me more details?